### ISSUE TITLE
Error running istanbul check-coverage --statements 90

### ISSUE THREAD

**Issue by leore** [gotwarlost/istanbul#192]
Getting this error:

/Users/bavidar/.nvm/v0.10.26/lib/node_modules/istanbul/lib/object-utils.js:58
                var line = statementMap[st].start.line,
                                           ^
TypeError: Cannot read property 'start' of undefined

Has anyone solves this problem?

**Comment by guyellis** [comment]
@leore - Is this still an issue for you? I think that it's an easy fix that I can do an submit a PR for if it's still an issue for you or someone else.

**Comment by chengyin** [comment]
I hit this problem recently. What solutions do you have in mind if you don't mind me asking, @guyellis?

**Comment by guyellis** [comment]
https://github.com/gotwarlost/istanbul/blob/master/lib/object-utils.js#L59

@chengyin I was going to skip that section of code if `statementMap[st]` was falsey. Thought that would be better than having it fail.

Are you hitting this error as well?

**Comment by gotwarlost** [comment]
This is most probably because it is trying to process a Json file that doesn't have coverage info. The error message is terrible and doesn't indicate which gold had the problem. If you want to submit a pr, you should probably focus on improving the message and / or skipping files that don't look like a coverage json

**Comment by chengyin** [comment]
I did some investigation and this is what I got for our specific case:

The issue is caused by a mismatch between `statements` and `statementsMap`. The length of the two didn’t agree. One recorded more statements than the other for the same file.

I dug deeper and in our codebase, this is because we are using webpack [inject-loader](https://github.com/plasticine/inject-loader) for our unit tests. If the injected version and the original version both get required, then there are two files registered to the same filename in Istanbul. The extra code for injection threw Istanbul off and eventually caused the exception.

Now that’s an issue that Istanbul doesn’t and probably shouldn’t deal with. There are multiple places in Istanbul runs into exceptions because of this. I agree better error message would help.

I plan to fork inject-loader and add Istanbul ignore comments for the injection related code. A [different inject loader](https://github.com/LeanKit-Labs/amd-inject-loader#options) does this.

**Comment by guyellis** [comment]
@chengyin I agree with @gotwarlost - would be great to also submit a PR for istanbul that did one or more of:
* improved the message 
* skip files that don't look like coverage
* and based on your use case also
  * checked that the lengths of statements and statementsMap were the same or printed an error
  * or did some checking while iterating through each one so that missing properties weren't use as in my previous comment.

As per my previous "fix" idea my feeling is that istanbul should still try and work and give a partially good result if not complete one while printing out error messages about what it can't do while processing.

i.e. given a choice between:
* complete failure with no result and a good error message
  * AND
* partial failure with some result and a good error message
I'd take the latter

**Comment by ianribas** [comment]
Just so it may help others with the same error message, I had the same problem and it was a similar situation as the one reported by @chengyin: the same file with two different coverage information.

As we use `broserify`, the first was coming from a browserify `transform` (with `browserify-istanbul`) and the other from the `karma-coverage` preprocessor. Turns out, of course, the second is unnecessary and was causing problems when generating the report.

**Comment by silkentrance** [comment]
@gotwarlost since this is basically a mistake in the build process and also related to an external library, we should close this for good, don't you think?

**Comment by silkentrance** [comment]
@gotwarlost or create an FAQ from it?
